src/system/SystemClock.h

```diff
@@ -279,6 +279,66 @@ class ClockBase
      *                                      current time.
      */
     virtual CHIP_ERROR SetClock_RealTime(Microseconds64 aNewCurTime) = 0;
+
+    /**
+     * This function returns the last known good time (set via the call to SetClock_LastKnownGoodTime()) API.
+     *
+     * @param[out] aTime                 The last known good time, expressed as Unix time scaled to microseconds.
+     *
+     * @retval #CHIP_NO_ERROR            If the method succeeded.
+     * @retval #CHIP_ERROR_INVALID_TIME  If the "last known good time" was not set.
+     */
+    virtual CHIP_ERROR GetClock_LastKnownGoodTime(Microseconds64 & aTime)
+    {
+        if (mLastKnownGoodTime == Microseconds64(0))
+        {
+            return CHIP_ERROR_INVALID_TIME;
+        }
+        aTime = mLastKnownGoodTime;
```

Comment:
I would not expect mLastKnownGoodTime to "run backwards".  That is, I'd expect it to only be updated if aTime > mLastKnownGoodTime.